loadbalancer/algorithm.go

```diff
@@ -121,10 +117,29 @@ func withFadeIn(rnd *rand.Rand, ctx *routing.LBContext, wi []int, wf []float64,
 	)
 
 	if rnd.Float64() < f {
-		return ctx.Route.LBEndpoints[choice]
+		return ep[choice]
+	}
+	for i := 0; i < len(ep); i++ {
+		if _, fadingIn := fadeInState(now, ctx.Route.LBFadeInDuration, ep[i].Detected); !fadingIn {
+			notFadingIndexes = append(notFadingIndexes, i)
+		}
 	}
 
-	return shiftToRemaining(rnd, ctx, wi, wf, now)
+	switch a := algo.(type) {
+	case *roundRobin:
+		return shiftToRemaining(a.rnd, ctx, notFadingIndexes, a.fadingWeights, now)
+	case *random:
+		return shiftToRemaining(a.rand, ctx, notFadingIndexes, a.fadingWeights, now)
+	case consistentHash:
```

Comment:
why this is no pointer but the others are?
I think all of these should have a similar type, so either ptr or value.

Reply:
Ah I just did it like this because the random and roundRobin algorithms implemented Apply as pointers and consistentHash didn't. I can change it so consistentHash uses pointers instead though :+1: